kore/src/Kore/Simplify/Condition.hs

```diff
@@ -97,7 +97,7 @@ simplify SubstitutionSimplifier{simplifySubstitution} sideCondition original = d
     return result
   where
     limit :: Int
-    limit = 4
+    limit = 1000
```

Comment:
That looks like a big change... do we _need_ that much?
Could this be an option to `kore-exec`?

Reply:
This could be what's taking so long on the KEVM proof. I see a message saying "retried 1000 times" in the output.

Reply:
The really long KEVM proof has a bunch of output like this:

```
kore-exec: [1137864347] Warning (WarnUnsimplifiedPredicate):                                                                                                                                                         
    Predicate not simplified after 1000 rounds.                                                                                                                                                                      
    Original predicate:                                                                                   
        /* Spc */                                                                                                                                                                                                    
        \and{_}(                                                                                          
            /* Spc */                                                                                     
            \equals{SortBool{}, _}(                                                                       
                /* Fn Sfc */                                                                                                                                                                                         
                Lbl'Unds'andBool'Unds'{}(                                                                 
                    /* Fn Sfc */                                                                                                                                                                                     
                    Lbl'Unds'andBool'Unds'{}(                                                                                                                                                                        
                        /* Fn Sfc */                                                                                                                                                                                 
                        Lbl'Unds'andBool'Unds'{}(                                                                                                                                                                    
                            /* Fn Sfc */                                                                                                                                                                             
                            Lbl'Unds'andBool'Unds'{}(                                                     
  
```

Reply:
The 1000 is of course just a symptom, not the actual cause. Somehow this proof hits a case that cannot be simplified or takes a lot of iterations to simplify.
The above, a giant `\equals ( \and(\and(\and(...`, does not show what is actually `and`-ed there, unfortunately, but the `\equals` at the top might just compare to `true` in the end?
Anyway, great that we have a setup to reproduce it!

Reply:
Yes, that's true. I bet it's something to do with the simplified attribute again. I'll take a look.

> That looks like a big change... do we need that much?

We used to only need so little because the Ceil simplifier would attempt to fully simplify everything under the Ceil. I moved those re-traversals to the top level and I didn't know how many I'd need. It might be large, but if something doesn't get simplified after 1000 iterations it signals that something is definitely wrong. I can change it to be smaller if you think it does more bad than good right now.